pyvo/utils/tests/test_prototype.py

```diff
@@ -0,0 +1,149 @@
+import warnings
+from copy import deepcopy
+from unittest import mock
+
+import pytest
+
+from pyvo.utils import prototype_feature, activate_features, prototype
+from pyvo.utils.prototype import PrototypeError, PrototypeWarning, Feature
+
+
+@pytest.fixture
+def prototype_function(features):
+    features({
+        'my-feature': Feature('my-feature', url='http://somewhere/else')
+    })
+
+    @prototype_feature('my-feature')
+    def i_am_prototype(arg):
+        arg('called')
+
+    return i_am_prototype
+
+
+@pytest.fixture
+def features():
+    previous_available = deepcopy(prototype.features)
+    prototype.features.clear()
+
+    def add(features):
+        prototype.features.update(features)
+
+    yield add
+
+    prototype.features.clear()
+    prototype.features.update(previous_available)
+
+
+def test_feature_turned_off_by_default(prototype_function):
+    with pytest.raises(PrototypeError) as e:
+        prototype_function(None)
+
+    assert str(e.value) == 'i_am_prototype is part of a prototype feature (my-feature) that has not been activated. ' \
+                           'For more information please visit http://somewhere/else'
+
+
+def test_activate_feature(prototype_function):
+    probe = mock.Mock()
+
+    activate_features('my-feature')
+
+    try:
+        prototype_function(probe)
+    except Exception as exc:
+        assert False, f"Should not have raised {exc}"
```

Comment:
Why this try/except?

Reply:
It's just to make the intent more expressive. The function is supposed to throw an exception unless a condition is met, this might better express that when reviewing the code.